spd/utils/component_utils.py

```diff
@@ -96,11 +76,37 @@ def sample_uniform_k_subset_routing_masks(
     return {mod: perms[i] < k_modules_to_route for i, mod in enumerate(module_names)}
 
 
+RoutingType = Literal["uniform_k-stochastic", "all"]
+"""How to choose which (batch,) or (batch, seq_len) positions to route to components or target.
+
+uniform_k-stochastic:
+    for each position, sample k from [1, n_modules], then route to components for k out of
+    `n_modules` modules
+all:
+    use components for all positions
+given:
+    use the given routing masks
+"""
+
+
+def calc_routing_masks(
+    routing: RoutingType,
+    leading_dims: tuple[int, ...],
+    module_names: list[str],
+    device: torch.device | str,
+) -> RoutingMasks:
+    match routing:
+        case "all":
+            return "all"
+        case "uniform_k-stochastic":
+            return sample_uniform_k_subset_routing_masks(leading_dims, module_names, device)
```

Comment:
I think I'd prefer just not having this function and doing the if statement explicitly at call sites. Reasons:
1. The function does nothing except in one specific case (uniform_k-stochastic)
2. 3/4 arguments to the function are only used in that one case.
3. There are only 2 call sites to this.